eogrow/pipelines/export_maps.py

```diff
@@ -114,16 +128,46 @@ def create_maps(self, feature_name: str, patch_names: List[str]) -> None:
 
             # manually make subfolder, otherwise things fail on S3 in later steps
             self.storage.filesystem.makedirs(fs.path.join(folder, subfolder), recreate=True)
-            self._create_single_map(folder, eopatch_list, subfolder, map_name)
+            self._create_single_map(folder, subfolder, map_name, eopatch_list)
 
-    def _create_single_map(self, folder: str, eopatch_list: List[str], subfolder: str, map_name: str) -> None:
-        """Creates a single map"""
-        make_local_copies = self.storage.is_on_aws() or self.config.force_local_copies
+    def _create_single_map(self, folder: str, subfolder: str, map_name: str, eopatch_list: List[str]) -> None:
+        """Creates a single map
 
+        If necessary, copies the geotiffs to local storage. After the merging transfers the merged map to the
+        appropriate location and removes the per-eopatch tiffs.
+        """
         geotiff_names = [self.get_geotiff_name(name) for name in eopatch_list]
         geotiff_paths = [fs.path.join(folder, name) for name in geotiff_names]
         merged_map_path = fs.path.join(folder, subfolder, map_name)
 
+        temp_fs, sys_paths, sys_map_path = self._prepare_files(map_name, geotiff_names, geotiff_paths, merged_map_path)
+
+        self._merge_maps(sys_paths, sys_map_path)
+
+        if temp_fs is not None:  # copy to storage if a temporary FS was used
+            fs.copy.copy_file(temp_fs, map_name, self.storage.filesystem, merged_map_path)
+            temp_fs.close()
+
+        LOGGER.info("Merged map is saved at %s", get_full_path(self.storage.filesystem, merged_map_path))
+
+        # clean up eopatch tiffs
+        parallelize(
+            self.storage.filesystem.remove, geotiff_paths, workers=None, multiprocess=False, desc="Remove eopatch tiffs"
+        )
+
+    def _prepare_files(
+        self,
+        map_name: str,
+        geotiff_names: List[str],
+        geotiff_paths: List[str],
+        merged_map_path: str,
+    ) -> Tuple[Optional[TempFS], List[str], str]:
+        """Returns system paths that can be used to merge maps.
+
+        If required files are copied locally and a temporary filesystem object is returned.
+        """
+        make_local_copies = self.storage.is_on_aws() or self.config.force_local_copies
+
         if make_local_copies:
             temp_fs = TempFS(identifier="_merge_maps_temp")
```

Comment:
If I remember correctly, the issue in the past was that this merging took a reeeeeally long time if a lot of tiffs were being merged from S3, so in that case it was better to first copy the tiffs to a local dir, then merge, and then copy to the target place. 

If I understand correctly, this is now not the case and could potentially be problematic again, right? (assuming I correctly remember the issue)

Reply:
That is still a potential problem. If i did everything correctly then the files should still be copied when needed.

Reply:
Ah, I missed the part where you put this into a specific function. The logic seems to stay the same, yes.